core/domain/story_domain.py

```diff
@@ -656,6 +656,26 @@ def require_valid_title(cls, title):
         if title == '':
             raise utils.ValidationError('Title field should not be empty')
 
+    def get_acquired_skill_ids_for_node_ids(self, node_ids):
+        """Returns the acquired skill ids of the nodes having the given
+        node ids.
+
+        Args:
+            node_ids: list(str). The list of IDs of the nodes inside
+            the story.
+
+        Returns:
+            list(str)|None. The list of acquired skill ids for the
```

Comment:
The union of the lists of ...

Also this docstring isn't written that clearly. What if some node IDs have nodes linked to them and some don't? You probably want to treat those as "empty list"? In which case the return value is _always_ a list(str) (possibly empty)?

Also, in general, if None is a possible return value then you should have a backend test for that case.

Reply:
This is not a union of lists. It is only one list containing all acquired skill ids.

Reply:
I'm using "union" here in the context of "set union". A union of two sets is still a set.

Reply:
Oh ok got it.